liblwgeom/lwline.c

```diff
@@ -607,3 +607,68 @@ POINTARRAY* lwline_interpolate_points(const LWLINE *line, double length_fraction
     return opa;
 }
 
+extern LWPOINT* lwline_interpolate_point_3d(const LWLINE *line, double distance)
+{
+	double length, slength, tlength;
+	POINTARRAY *ipa;
+	POINT4D pt;
+	int nsegs, i;
+	LWGEOM* geom = lwline_as_lwgeom(line);
+	int has_z = lwgeom_has_z(geom);
+	int has_m = lwgeom_has_m(geom);
+	ipa = line->points;
+
+	/* Empty.InterpolatePoint == Point Empty */
+	if ( lwline_is_empty(line) )
+	{
+	  return lwpoint_construct_empty(line->srid, has_z, has_m);
+	}
+
+	/* If distance is one of the two extremes, return the point on that
+	 * end rather than doing any expensive computations
+	 */
+	if ( distance == 0.0 || distance == 1.0 )
```

Comment:
I'd suggest using `FP_EQUALS` instead of comparing doubles directly.

Reply:
Is it? What about `-0.0`? Or `0.9999999999999983` which is, according to our internal standards, 1.0?

Reply:
okay, then we need to change this function and the function right above it to match. :)